bot/exts/moderation/infraction/infractions.py

```diff
@@ -664,6 +645,29 @@ async def cog_command_error(self, ctx: Context, error: Exception) -> None:
                 await ctx.send(str(error.errors[0]))
                 error.handled = True
 
+    @commands.Cog.listener()
+    async def on_member_join(self, member: Member) -> None:
+        """
+
+        Apply active timeout infractions for returning members.
+
+        This is needed for users who might have had their infraction edited in our database but not in Discord itself.
+        """
+        active_timeouts = await self.bot.api_client.get(
+            endpoint="bot/infractions",
+            params={"active": "true", "type": "timeout", "user__id": member.id}
+        )
+
+        if active_timeouts:
+            timeout_infraction = active_timeouts[0]
+            expiry = arrow.get(timeout_infraction["expires_at"], tzinfo=UTC).datetime
+
+            reason = f"Applying active timeout for returning member: {timeout_infraction['id']}"
+
+            async def action() -> None:
+                await member.edit(timed_out_until=expiry, reason=reason)
+            await self.reapply_infraction(timeout_infraction, action)
```

Comment:
The only thing I *miiiiight* change would be to check if `member.timed_out_until`  is not same as the expiry time in the database before applying the timeout. This could potentially save an API call in the handful of scenarios when a user under a timeout rejoins the server and the timeout duration is unchanged (*although*, it is more likely that in the event a timed-out user does rejoin the server, the duration has not been edited).
Again, this case does not happen that often, and even if it did, that extra API call is not that big of a deal. So, you have my approval for merging this PR.

Reply:
I'll accept this only because your houseplants are talking good of you about your Morse code teaching methodology.

Implemented [here](https://github.com/python-discord/bot/compare/eba77402114dab719c5b9f45d0478aa19d9f9fcf..129705a5b195b36e5b2c64fa2beb5822cefeb9de)